solver/src/settlement.rs

```diff
@@ -151,6 +188,20 @@ impl Settlement {
         self.encoder.trades()
     }
 
+    /// Returns an iterator of all executed trades.
+    pub fn executed_trades(&self) -> impl Iterator<Item = TradeExecution> + '_ {
```

Comment:
Is this the place where the `import as _` is used? If so it seems its only purpose is here, what is the advantage or reason for import `U256Ext as _`?

Reply:
`use Trait as _` allows you to use trait methods without importing the trait name into scope. Here the `'_` (notice the `'`) is a lifetime - specifically the lifetime of `&self`. This means I return an iterator which lives as long as the `&self` reference that was passed into the method.

It is needed because a closure that is being returned captures `&self`.